wfl/Database/MongoDBManager.cs

```diff
@@ -8,55 +8,31 @@
 
 namespace Symphono.Wfl.Database
 {
-    public class MongoDBManager: IDBManager
+    public class MongoDBManager<T>: IDBManager<T> where T: IEntity
     {
         private static string connectionString;
         public static MongoClient client;
         public static IMongoDatabase db;
+        private string collectionName { get; }
```

Comment:
With C#6, they introduced a `readonly` keyword which is perfect for variables like this that are assigned in the constructor and should never be changed. I'd recommend using that whenever it makes sense. That way someone can't change the value using reflection.